contracts/truefi2/TrueCreditAgency.sol

```diff
@@ -268,10 +273,35 @@ contract TrueCreditAgency is UpgradeableClaimable {
         pool.token().safeTransfer(msg.sender, amount);
     }
 
-    function repay(ITrueFiPool2 pool, uint256 amount) external {
-        pool.token().safeTransferFrom(msg.sender, address(this), amount);
-        pool.token().safeApprove(address(pool), amount);
-        pool.repay(amount);
+    function payInterest(ITrueFiPool2 pool) external {
+        uint256 accruedInterest = interest(pool, msg.sender);
+        _payInterestWithoutTransfer(pool, accruedInterest);
+        _repay(pool, accruedInterest);
+    }
+
+    function repay(ITrueFiPool2 pool, uint256 amount) public {
+        uint256 currentDebt = borrowed[pool][msg.sender];
+        uint256 accruedInterest = interest(pool, msg.sender);
+        if (amount < accruedInterest) {
+            _payInterestWithoutTransfer(pool, amount);
+            return;
```

Comment:
> If you return here, then you haven't transferred any tokens yet.

How did I miss it 😓

Reply:
The reason I created `_payInterestWithoutTransfer ` is to not write the same lines of code twice. In case of principal repayment it happens only once in this contract.